hermes-client/src/main/java/pl/allegro/tech/hermes/client/HermesClientBasicRetryCondition.java

```diff
@@ -1,13 +1,18 @@
 package pl.allegro.tech.hermes.client;
 
+import java.net.SocketTimeoutException;
+import java.util.Optional;
 import java.util.function.Predicate;
 
 import static java.net.HttpURLConnection.HTTP_CLIENT_TIMEOUT;
 
 public class HermesClientBasicRetryCondition implements Predicate<HermesResponse> {
     @Override
     public boolean test(HermesResponse response) {
-        return response == null || (isClientTimeoutOrServerError(response) || isFailedExceptionally(response));
+        return response == null
+            || isClientTimeoutOrServerError(response)
+            || isFailedExceptionally(response)
+            || isSocketTimeoutException(response.getFailureCause());
```

Comment:
To keep the code more common, maybe we can change `isSocketTimeoutException` param to the `HermesResponse` ? WDYT?

Reply:
I agree